deeplay/components/dcgan/dcgan.py

```diff
@@ -0,0 +1,464 @@
+from typing import List, Optional, Literal, Any, Sequence, Type, overload, Union
+
+from ... import (
+    DeeplayModule,
+    Layer,
+    LayerList,
+    Sequential,
+    LayerActivation,
+    LayerActivationNormalization,
+)
+
+import torch
+import torch.nn as nn
+
+from deeplay import LayerList
+
+
+class ConvBlock(DeeplayModule):
+    """
+    Convolutional block for DCGAN discriminator. It consists of a 2D Convolutional layer, a Batch Normalization layer and a LeakyReLU activation function.
+    """
+
+    def __init__(
+        self,
+        in_channels,
+        out_channels,
+        kernel_size,
+        stride,
+        padding,
+        batch_norm=True,
+    ):
+        super().__init__()
+        self.blocks = LayerList()
+        self.blocks.append(
+            LayerActivationNormalization(
+                Layer(
+                    nn.Conv2d,
+                    in_channels,
+                    out_channels,
+                    kernel_size,
+                    stride,
+                    padding,
+                    bias=not batch_norm,
+                ),
+                Layer(nn.BatchNorm2d, out_channels)
+                if batch_norm
+                else Layer(nn.Identity),
+                Layer(nn.LeakyReLU, 0.2),
+            )
+        )
+
+    def forward(self, x):
+        for block in self.blocks:
+            x = block(x)
+        return x
```

Comment:
Should be defined in components/blocks, and subclass `LayerActivationNormalization`. Something like

```py
class ConvBlock(LayerActivationNormalization):
    def __init__(
        self,
        in_channels,
        out_channels,
        kernel_size,
        stride,
        padding,
        activation=None,
        normalization=None,
    ):
    super().__init__(
         layer=Layer(Conv2d, kernel_size=kernel_size, ...),
         activation=activation or Layer(ReLU),
         normalization=normalization or Layer(BatchNorm2d, out_channels)
    )

Reply:
Fixed this now. However, it is still at dcgan/dcgan.py. Do you want me to move it somewhere else?